pyvista/utilities/reader.py

```diff
@@ -2330,6 +2332,66 @@ class GIFReader(BaseReader):
     _class_reader = _GIFReader
 
 
+class XdmfReader(BaseReader, PointCellDataSelection):
+    """XdmfReader for .xdmf files.
+
+    Parameters
+    ----------
+    path : str
+        Path of the XDMF file to read.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> from pyvista import examples
+    >>> filename = examples.download_dolfinx_mesh(load=False)
+    >>> reader = pyvista.get_reader(filename)
+    >>> filename.split("/")[-1]  # omit the path
+    'mesh.xdmf'
+    >>> mesh = reader.read()
+    >>> mesh.plot()
+
+    """
+
+    _class_reader = staticmethod(_vtk.lazy_vtkXdmfReader)
+
+    @property
+    def number_grids(self):
+        """Get information about grids.
+
+        Returns
+        -------
+        int
+
+        """
+        return self.reader.GetNumberOfGrids()
+
+    @property
+    def number_point_arrays(self):
+        """Return the number of point arrays.
+
+        Returns
+        -------
+        int
+
+        """
+        return self.reader.GetNumberOfPointArrays()
+
+    @property
+    def point_array_names(self):
+        """Return the list of all point array names.
+
+        Returns
+        -------
+        list[str]
+
+        """
+        return [self.reader.GetPointArrayName(i) for i in range(self.number_point_arrays)]
+
+    def set_active_time_value(self, time_value):  # noqa: D102
```

Comment:
Does this Reader not support the `TimeReader` class inheritance?  It looks like it should based on my quick glance at the vtk docs.  If not, it will be confusing that we still define this method.

Reply:
@MatthewFlamm I think it should be supported. I will investigate whether the method can be supported. If not, I will use it to generate an error stating that it is unsupported.

Reply:
I did some digging and it does not seem to support `TimeReader` inheritance, although Paraview can see the time step information.  In the tests, it uses the vtk pipeline in some manner to get the time steps.  However, I'm unsure how this would mesh with PyVista's current 'pipeline-less' approach.

https://gitlab.kitware.com/vtk/vtk/-/blob/v9.2.0/IO/Xdmf2/Testing/Cxx/TestTemporalXdmfReaderWriter.cxx#L40-48

I'm fine for just putting in a note in the docstring that we currently can't inspect the time values for this reader.

Reply:
Thanks for your search I added it at https://github.com/pyvista/pyvista/pull/2839/commits/ab2299d9855de0de34243410aaf2dce56bc01f50.